numpy/core/_methods.py

```diff
@@ -56,16 +59,23 @@ def _any(a, axis=None, dtype=None, out=None, keepdims=False):
 def _all(a, axis=None, dtype=None, out=None, keepdims=False):
     return umr_all(a, axis, dtype, out, keepdims)
 
-def _count_reduce_items(arr, axis):
-    if axis is None:
-        axis = tuple(range(arr.ndim))
-    if not isinstance(axis, tuple):
-        axis = (axis,)
-    items = 1
-    for ax in axis:
-        items *= arr.shape[mu.normalize_axis_index(ax, arr.ndim)]
+def _count_reduce_items(arr, axis, keepdims=False, where=True):
+    if isinstance(where, bool):
+        # no boolean mask given, calculate items according to axis
```

Comment:
This doesn't handle `mask = False` correctly.

Reply:
I first thought about leaving it out because setting `where=False` would basically mean to intentionally calculate the mean or variance of an empty slice. But you are right, I should return zero in this case. The update is in the next version.

Reply:
This comment confused me a bit, is this still required. What if `where=True` is provided when calling np.mean.

Reply:
The default would be to have `where=True`, however in this earlier version, I did not handle `where=False` specifically, because I thought using `where=False` would not make sense in general. However to ensure correct behavior even for this case (`where=False` means attempting to calculate the mean of an empty slice), I added a branch.
Does this make it clearer, @anirudh2290 ?

Reply:
okay, thanks @sgasse ! i guess i got confused probably because I didnt first understand that "no boolean mask given" is same as "where=True has the same behavior as when no boolean mask is given". This also makes me think if this will cause a failure when where=None (probably a not so common use case).

Reply:
I don't think where=None is legal anyway, but perhaps worth checking

Reply:
@anirudh2290 `where=None` and `where=False` indeed lead to the divide-by-zero warning and `nan` as result. I am not sure this is how it should be, e.g. `arr.sum(where=None)` returns zero without any warning. Should I adapt `mean` in that way or would it clutter the function too much for catching an illegal input?

Reply:
That behavior sounds correct to me. Mean is different to sum because it has to divide by the number of non-masked elements.